liblangutil/SourceLocation.h

```diff
@@ -49,6 +52,13 @@ struct SourceLocation
 
 	bool isEmpty() const { return start == -1 && end == -1; }
 
+	std::string text() const
+	{
+		assertThrow(source, SourceLocationError, "Requested text from null source.");
+		assertThrow(!isEmpty(), SourceLocationError, "Requested text from empty source location.");
+		return source->source().substr(start, end - start);
```

Comment:
Should we assert that ``start <= end`` and ``end <= source->source().length()`` or is that overly paranoid :-)?

Reply:
It's paranoid, but doesn't hurt ;) What would the message be? Something like `Requested text from an invalid source location.`?

Reply:
Or just invalid source location?

Reply:
updated